### PR TITLE
feat: osmosis delegation support

### PR DESCRIPTION
**Pull request by elmutt**
## Description

Add delegation support for osmosis

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Osmosis bounty

## Risk

Some risk that cosmos delegation could be affected by this

## Testing

- Make sure cosmos staking still works

- No need to fully test osmosis staking as its behind a feature flag (we know the apr is showing as 0 and will fix it from unchained)

### CODE REVIEW

File: src/state/slices/portfolioSlice/selectors.ts
Comment (DaoDev44) [inline src/state/slices/portfolioSlice/selectors.ts:101]: maybe use case statement to check?

**Review by DaoDev44 — DISMISSED** [review]

File: src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx
Comment (0xdef1cafe) [inline src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx:42]: you can define this adapter once to avoid casting three times

you can also define the payload once, and just use this switch statement to spread it onto actions

will eliminate most of these lines here

File: src/state/slices/portfolioSlice/selectors.ts
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/selectors.ts:102]: can you please do an `F2` (rename symbol) and make this `SHAPESHIFT_COSMOS_VALIDATOR_ADDRESS` now, as previously we only had one, but now we have OSMO it's not clear what this other one is

File: src/state/slices/validatorDataSlice/const.ts
Comment (0xdef1cafe) [inline src/state/slices/validatorDataSlice/const.ts:1]: do the rename symbol here and it will propagate everywhere

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
small requested changes but conceptually this is good

File: src/pages/Defi/hooks/useEarnBalances.tsx
Comment (0xApotheosis) [inline src/pages/Defi/hooks/useEarnBalances.tsx:42]: We can now use the `cosmosAssetId` and `osmosisAssetId` constants from `shapeshiftoss/caip` instead of stings in the two instances in this file.

File: src/plugins/cosmos/components/modals/Staking/views/GetStarted/GetStarted.tsx
Comment (0xApotheosis) [inline src/plugins/cosmos/components/modals/Staking/views/GetStarted/GetStarted.tsx:22]: Same here, we can use the imported constants.

File: src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx
Comment (0xApotheosis) [inline src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx:43]: I'd love to see us change this to an IIFE instead of mutating `result`, but non-blocking.

File: src/state/slices/validatorDataSlice/const.ts
Comment (0xApotheosis) [inline src/state/slices/validatorDataSlice/const.ts:1]: We use `constants.ts` as the filename for storing constants everywhere else - can we please update this to keep it consistent?

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Some small requested changes, otherwise looks good.

Tested for regressions and I was able to stake and unstake ATOM.

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
LGTM overall, a few requested changes/questions

File: src/state/slices/portfolioSlice/selectors.ts
Comment (elmutt) [reply src/state/slices/portfolioSlice/selectors.ts:100]: How about `getShapeshiftValidatorFromAccountSpecifier` ?

File: src/state/slices/validatorDataSlice/validatorDataSlice.ts
Comment (elmutt) [reply src/state/slices/validatorDataSlice/validatorDataSlice.ts:68]: intersection makes sense to me for this case

**Comment by 0xdef1cafe** [comment]
adding blocked until unchained stuff is merged

File: src/state/slices/portfolioSlice/selectors.ts
Comment (gomesalexandre) [reply src/state/slices/portfolioSlice/selectors.ts:100]: Sounds good to me!

File: src/state/slices/validatorDataSlice/validatorDataSlice.ts
Comment (0xApotheosis) [reply src/state/slices/validatorDataSlice/validatorDataSlice.ts:68]: Or some type that is `cosmossdk.cosmos.ChainAdapter | cosmossdk.osmosis.ChainAdapter`?

To @gomesalexandre point, it's never both, but always one or the other.

**Comment by elmutt** [comment]
closing to re-open a new PR due to massive merge conflicts

**Pull request closed without merging**
